x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/structurefinder/DelimitedTextStructureFinderFactory.java

```diff
@@ -89,4 +105,26 @@ public TextStructureFinder createFromSample(
             timeoutChecker
         );
     }
+
+    public TextStructureFinder createFromMessages(
+        List<String> explanation,
+        List<String> messages,
+        TextStructureOverrides overrides,
+        TimeoutChecker timeoutChecker
+    ) throws IOException {
+        // DelimitedTextStructureFinderFactory::canCreateFromMessages already
+        // checked that every line contains a single valid delimited message,
+        // so we can safely concatenate and run the logic for a sample.
+        String sample = String.join("\n", messages);
+        return DelimitedTextStructureFinder.createFromSample(
+            explanation,
+            sample,
+            "UTF-8",
+            null,
+            csvPreference,
+            trimFields,
+            overrides,
+            timeoutChecker
+        );
```

Comment:
Maybe we should also pass in an extra argument here to force the value of `isHeaderInText` on line 65, because if we have a series of documents each with 1 CSV row in them then we know that the header row containing column names is not in the data and any column names must be autogenerated or come from overrides. 99% of the time the algorithm will probably work this out, but if it doesn't then the user will be very confused, so it's best to override that detection.

Reply:
I've set
```
request.setHasHeaderRow(false);
```
in the `RestFindStructureArgumentsParser` for the `FindFieldStructureAction.Request` and `FindMessageStructureAction.Request`. Then this setting will propagate via the `TextStructureOverrides` in here.